symengine/visitor.h

```diff
@@ -90,6 +90,30 @@ void preorder_traversal_stop(const Basic &b, StopVisitor &v);
 void postorder_traversal_stop(const Basic &b, StopVisitor &v);
 void preorder_traversal_local_stop(const Basic &b, LocalStopVisitor &v);
 
+class HasBasicVisitor : public BaseVisitor<HasBasicVisitor, StopVisitor>
+{
+protected:
+    Ptr<const Basic> looking_for_;
+    bool has_;
+
+public:
+    HasBasicVisitor(Ptr<const Basic> x) : looking_for_(x) {}
```

Comment:
We should raise a NotImplementedError when `x` is an associative op like `Mul, Add, Or` since we don't check for subexpressions.

Reply:
But if we want to check for subexpressions?

Reply:
alright then, I'll try to filter those then. I'm not quite sure what the best approach would be, `is_a_Atom` is not quite right since e.g. Infty has non-empty `get_args` vector. So I guess I should spell out our type codes. What about `Pow`? It doesn't contain arbitrary number of arguments, so it's then ok to match against it?

Reply:
Yeah, but bug or not, that's how `Infty` is implemented (it returns `_direction` which gives the sign e.g. +Inf vs. -Inf). I guess a changing that should be a different PR, and I'm not sure what backcomp issues may arise (my gut feeling is that it might be a few...)